counsel.el

```diff
@@ -2535,6 +2544,14 @@ regex string."
 
 (defvar counsel-ag-command nil)
 
+(defvar counsel-ag-look-around-p t
```

Comment:
If this is not specific to `counsel-ag`, then it should be given a different name, e.g. based on "search" or the more general term "grep". Also, the `-p` suffix convention applies only to predicates (functions), not variables, in Emacs.

Reply:
This was discussed more at the bottom, but the way Grep style functions are implemented on counsel are as wrappers around `counsel-ag`. This matched the namespace for other functions and variables such as `counsel-ag-command`, and `counsel-ag-function` which are used by all Grep functions. I considered some refactoring to remove the wrappers and instead use common subroutines. But I also don’t want this PR to go out of scope.

Reply:
The `-p` suffix is used in variables. I did a search in describe variable for `-p$` and found about a hundred matches. Granted some of them are variables that hold functions, but others are not. They come from all packages, including magit, helm, lispy, gnus, c-mode,  yas, and org.

Reply:
> The `-p` suffix is used in variables. I did a search in describe variable for `-p$` and found about a hundred matches. Granted some of them are variables that hold functions, but others are not. They come from all packages, including magit, helm, lispy, gnus, c-mode, yas, and org.

That it's done by other packages doesn't make it right. The Emacs convention is to give this suffix only to predicate functions.

Reply:
> This was discussed more at the bottom, but the way Grep style functions are implemented in counsel are as wrappers around `counsel-ag`.

Exactly. It's an *implementation detail* (and a kludge at that) that shouldn't affect what we name new definitions.

> This matched the namespace for other functions and variables such as `counsel-ag-command`, and `counsel-ag-function` which are used by all Grep functions.

This is still an implementation detail. User options are for users, not programmers. Search commands are search (or grep) commands, not `counsel-ag` commands.

Reply:
True, but `counsel-ag-command` and `counsel-ag-look-around` are not user options. They are explicitly overridden by the `counsel-ag` wrappers like `counsel-pt` and `counsel-rg`, and therefore they don't have any effect if set by the user.

Reply:
> True, but `counsel-ag-command` and `counsel-ag-look-around` are not user options. They are explicitly overridden by the `counsel-ag` wrappers like `counsel-pt` and `counsel-rg`, and therefore they don't have any effect if set by the user.

That they're plain variables doesn't mean we should give them an inaccurate name when we have the choice.

Reply:
It follow the naming convention in place (`counsel-ag-command` and `counsel-ag-function` ). 

I take your point that these names are misleading, and I agree that we should change them. But I still think that should be a separate PR. I don't want this to become a giant refactoring PR.

Reply:
> It follow the naming convention in place

It's not really a convention. Using "search" or "grep", on the other hand, is.

> I take your point that these names are misleading, and I agree that we should change them. But I still think that should be a separate PR. I don't want this to become a giant refactoring PR.

I'm not asking you to rename existing definitions. I'm asking you to not introduce new definitions with misleading names.

> For this PR, it will keep it simple to follow the design pattern already used.

Calling it a "design pattern" is an overstatement if I ever saw one. ;)

Reply:
Okay, I take your point. I fixed it.